lib/eco/treemanager.py

```diff
@@ -683,6 +683,8 @@ def get_max_version(self):
 
     def key_ctrl_z(self):
         self.log_input("key_ctrl_z")
+        if not self.undo_snapshots and self.get_max_version() > 1:
```

Comment:
If `undo_snapshots` is a list, it's probably clearer to say `if len(undo_snapshots) == 0`?